palettes.h

```diff
@@ -811,3 +814,51 @@ struct PaletteGroups {
 		}
 	}
 };
+
+// This is 100% registry-backed storage. Do not access it in hot paths.
+class PaletteRepository {
+public:
+	PaletteRepository();
+
+	void set_palette(const std::wstring &name, const PaletteData &data);
+	void remove_palette(const std::wstring &name);
+	std::optional<Palettes::Definition> get_palette(const std::wstring &name);
+	std::vector<Palettes::Definition> get_all_custom_palettes();
+
+private:
+	std::wstring serialize(const PaletteData &palette);
+	PaletteData deserialize(const std::wstring &serialized);
+
+	RegistryBackedMap m_map;
+};
+
+class PaletteGroupRepository {

```

Comment:
Originally, I was going to put support for multiple custom palette groups in this PR, but that made the UI almost incomprehensible, so I abandoned it for now. I'm keeping this class around because I might come back to it.